src/plugins/auctionHouseModule/AuctionHouseClient.ts

```diff
@@ -85,6 +93,68 @@ export class AuctionHouseClient {
       .getTask(loadListingOperation({ lazyListing, ...options }));
   }
 
+  bid(input: WithoutAH<CreateBidInput>): Task<CreateBidOutput & { bid: Bid }> {
+    return new Task(async (scope) => {
+      const output = await this.metaplex
+        .operations()
+        .execute(createBidOperation(this.addAH(input)), scope);
+      scope.throwIfCanceled();
+
+      try {
+        const bid = await this.findBidByAddress(output.buyerTradeState).run(
+          scope
+        );
+        return { bid, ...output };
+      } catch (error) {
+        // Fallback to manually creating a bid from inputs and outputs.
+      }
+
+      scope.throwIfCanceled();
+      const lazyBid: LazyBid = {
+        model: 'bid',
+        lazy: true,
+        auctionHouse: this.auctionHouse,
+        tradeStateAddress: output.buyerTradeState,
+        bookkeeperAddress: input.printReceipt ? output.bookkeeper : null,
+        buyerAddress: output.wallet,
+        metadataAddress: output.metadata,
+        receiptAddress: input.printReceipt ? output.receipt : null,
+        purchaseReceiptAddress: null,
+        price: output.price,
+        tokens: output.tokens.basisPoints,
+        createdAt: toDateTime(new Date()),
+        canceledAt: null,
+      };
+
+      return {
+        bid: await this.loadBid(lazyBid).run(scope),
+        ...output,
+      };
+    });
+  }
+
+  findBidByAddress(
+    address: PublicKey,
+    options: { commitment?: Commitment; loadJsonMetadata?: boolean } = {}
```

Comment:
Could we change that to the input type omitting the address? I know you've tried to stay consistent with `findListingByAddress` but this was an oversight on my side. Could you do the same with the `findListingByAddress` please?

```suggestion
    options: Omit<FindBidByAddressInput, 'address'> = {}
```